### PR TITLE
Refactor chart code and fix hovercards on waterfall charts

### LINKED ISSUE #10: Add informative hover labels in charts

### LINKED ISSUE #284: Remove color attribute to the right of intradecile hovercards
For example, remove the redundant "Gain more than 5%" to the right of the hovercard here:
image

This is an issue for any chart that has custom hovercards and multiple colors, e.g. it'll be an issue with the waterfall charts being improved in #283.

I asked how to fix it in https://stackoverflow.com/q/69430653/1840471 and can follow up in the Plotly forums if I don't hear soon.

### LINKED ISSUE #287: Remove total_income and get_partial_funding functions
These appear to be artifacts of the old waterfall code and are no longer used:

https://github.com/PolicyEngine/policyengine-uk/blob/9c4b996aac196375b81e535b14050837323b97a4/policy_engine_uk/populations/charts.py#L83-L89

https://github.com/PolicyEngine/policyengine-uk/blob/9c4b996aac196375b81e535b14050837323b97a4/policy_engine_uk/populations/charts.py#L106-L107

### PR DESCRIPTION
**Pull request by MaxGhenis**
Completes the fixing of #10 (fixes #10)

Also fixes #284, fixes #287

### CODE REVIEW

**Comment by MaxGhenis** [comment]
Now ready. Here are a few examples:
image
image
Same label for the whitespace:
image
Final hovercard matches the top-level label:
image

Household level:
image
image

Same label above and below x axis:
image

Household net income label matches the headline:
image

File: policy_engine_uk/utils/charts.py [hunk 1]
```diff
@@ -15,7 +17,7 @@
 DARK_GREEN = "#558B2F"
 
 
-def format_fig(fig: go.Figure) -> dict:
+def formatted_fig_json(fig: go.Figure) -> dict:
```
Comment (MaxGhenis) [inline policy_engine_uk/utils/charts.py:20]: Does `json.loads(fig.to_json())` return a `dict` or `str`?

Comment (nikhilwoodruff) [reply policy_engine_uk/utils/charts.py:20]: `dict`: `json.loads` is `str -> dict`, `to_json` is `Plot -> str`

**Review by nikhilwoodruff — APPROVED** [review]
Much cleaner, thanks @MaxGhenis

**Pull request merged**